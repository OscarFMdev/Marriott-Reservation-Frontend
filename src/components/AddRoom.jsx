import React from 'react';
import AddRoomForm from './forms/AddRoomForm';

const AddRoom = () => (
  <div className="page">
    <AddRoomForm />
  </div>
);

export default AddRoom;
