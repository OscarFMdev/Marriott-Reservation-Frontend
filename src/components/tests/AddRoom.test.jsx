import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';

import AddRoom from '../AddRoom';
import store from '../../redux/configureStore';

describe('It displays the Add Rooom page title', () => {
  beforeEach(() => {
    render(
      <Provider store={store}>
        <Router>
          <AddRoom />
        </Router>
      </Provider>,
    );
  });
  test('Add Room', () => {
    expect(screen.queryAllByText(/Add Room/, { exact: true })).toHaveLength(2);
  });
});

test('Add Rooms renders correctly', () => {
  const tree = renderer.create(
    <Provider store={store}>
      <Router>
        <AddRoom />
      </Router>
    </Provider>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
