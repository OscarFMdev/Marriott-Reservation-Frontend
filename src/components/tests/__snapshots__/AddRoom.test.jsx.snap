// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Add Rooms renders correctly 1`] = `
<div
  className="page"
>
  <form
    className="_formContainer_6d2414"
    onSubmit={[Function]}
  >
    <div
      className="_hotel_6d2414"
    />
    <h1
      className="_header_6d2414"
    >
      Add Room
    </h1>
    <p
      className="_details_6d2414"
    >
       
      Welcome! Help this page grow by adding new rooms. You can add a new room by completing the following form:
    </p>
    <p
      style={
        {
          "color": "red",
        }
      }
    />
    <div
      className="_inputsContainer_6d2414 _center_6d2414"
    >
      <div
        className="_field_6d2414"
      >
        <label
          htmlFor="name"
        >
          <input
            id="name"
            onChange={[Function]}
            placeholder="Hotel room name"
            required={true}
            type="text"
            value=""
          />
        </label>
      </div>
      <div
        className="_field_6d2414"
      >
        <label
          htmlFor="image"
        >
          <input
            id="image"
            onChange={[Function]}
            placeholder="Image URL"
            required={true}
            type="text"
            value=""
          />
        </label>
      </div>
      <div
        className="_field_6d2414"
      >
        <label
          htmlFor="description"
        >
          <textarea
            id="description"
            onChange={[Function]}
            placeholder="Description"
            required={true}
            value=""
          />
        </label>
      </div>
      <div
        className="_field_6d2414"
      >
        <label
          htmlFor="price"
        >
          <input
            id="price"
            min={50}
            onChange={[Function]}
            placeholder="Price"
            required={true}
            type="number"
            value=""
          />
        </label>
      </div>
      <div
        className="_field_6d2414"
      >
        <label
          htmlFor="roomType"
        >
          <select
            id="roomType"
            onChange={[Function]}
            required={true}
            value=""
          >
            <option
              value=""
            >
              -- Select a room type --
            </option>
            <option
              value="Single"
            >
              Single
            </option>
            <option
              value="Double"
            >
              Double
            </option>
            <option
              value="Suite"
            >
              Suite
            </option>
          </select>
        </label>
      </div>
      <button
        type="submit"
      >
        Add Room
      </button>
    </div>
  </form>
</div>
`;
